src/liballoc/vec.rs

```diff
@@ -411,7 +411,10 @@ impl<T> Vec<T> {
     ///
     /// Violating these may cause problems like corrupting the allocator's
     /// internal data structures. For example it is **not** safe
-    /// to build a `Vec<u8>` from a pointer to a C `char` array and a `size_t`.
+    /// to build a `Vec<u8>` from a pointer to a C `char` array with length `size_t`,
+    /// or a `Vec<u16>` and its length. Because the allocator cares about the alignment.
+    /// The buffer was allocated with alignment 2 (for `u16`), but after
+    /// turning it into a `Vec<u8>` it'll be deallocated with alignment 1.
```

Comment:
This needs rephrasing to flow well with the rest of the paragraph. Since the alignment requirement is already stated above (second bullet point), I would also lean on that to make the note shorter. I'd suggest:

> For example, it is **not** safe to build a `Vec<u8>` from a pointer to a C `char` array and its length (as it was allocated differently), or from a `Vec<u16>` (as `u16` has different alignment than `u8`).